### ISSUE TITLE
Selection of placed building/foundations is inconsistent

### ISSUE THREAD

**Issue by Romboter** [brandon-ray/foxhole-facility-planner#12]
When trying to move a preplaced building it is quite difficult depending on the building.

Maint. tunnels are almost impossible to move or select once placed.
And for most other buildings it is hard to find the right spot to click to drag them around or modify their properties.

Possible solutions:

- Use the icon image in the middle as the selector for all buildings
- Use the colored outline as the selector
- Both of the above

Foundations have a related issue, when trying to select a building that is on top of foundations it sometimes selects the foundation and that is what gets dragged

Possible solutions:

- Add a lock position to the left menu when the foundation is selected
- Don't allow selecting of foundation if a building is on top

**Comment by jimdcunningham** [comment]
Oh wait does this still happen? My latest commit should've actually made selecting buildings much easier. I may have goofed though haha. I'm unsure about the foundation thing though.

**Comment by jimdcunningham** [comment]
Also, yeah I do plan on adding the ability to lock placed objects in the future just to help prevent booboos.

**Comment by Romboter** [comment]
The last change did make it easier to select and move objects, I was testing from my work pc earlier when I made this issue...but testing now from my home pc, chrome, and its working as expected.

I'll run the test again from my work pc tomorrow...incognito to rule out extensions.

Just thought of another idea for how to do the locking...double click the thing you want to toggle the lock against.

**Comment by brandon-ray** [comment]
I think it may be an issue with the browser zoom. I have noticed if I have accidentally zoomed the page (not the tool zoom) then selection can be weird and not work at all in some cases.

**Comment by Skjolbir** [comment]
Selection is straight up broken for me now, I cant select anything at all, whereas in the past it was inconsistent.
Please just let us select from a list.

**Comment by jimdcunningham** [comment]
> Selection is straight up broken for me now, I cant select anything at all, whereas in the past it was inconsistent. Please just let us select from a list.

Do you have any kind of scaling set on Windows or whichever OS you're using? Someone reported to us that they had theirs set 125% in 'Scale and Layout' inside of Windows and that setting it to 100% fixed their selection issues. Obviously we'll probably be able to fix this with math, but scaling / zoom by the browser / OS seems to be the culprit right now. Unless the rabbit hole goes deeper... rip

**Comment by Skjolbir** [comment]
150% as I am on 1440p, so its basically needed to be a comfortable monitor.
I have set it to 100% and found that fixed all selection issues. Thanks for the tip!         
Edit: I have found that 150% windows scaling, when paired with 67% browser scaling fixes the issue as well, which makes sense given the math (150 * 67 is about 100)

**Comment by jimdcunningham** [comment]
> 150% as I am on 1440p, so its basically needed to be a comfortable monitor. I have set it to 100% and found that fixed all selection issues. Thanks for the tip! Edit: I have found that 150% windows scaling, when paired with 67% browser scaling fixes the issue as well, which makes sense given the math (150 * 67 is about 100)

Just pushed up a commit to hopefully fix any issues with selecting buildings while browser / OS is zoomed / scaled. 👍
Also _should_ fix issues with the wrong building being selected when they're on top of each other.

Rail selection is still borked so Ray is taking care of that haha. 🥴

Edit: Also thanks for letting us know. We appreciate you guys reporting this. ❤️

**Comment by brandon-ray** [comment]
Fixed rails, everything should be good. :)

**Comment by Romboter** [comment]
The fix looks good, I was just doing a bunch of testing around it to see if I could break it.
Everything worked as expected!

The only thing I noticed was in relation to browser zoom.
If you browser zoom in, all the way to for example 500% it zooms on the canvas, but if you reset the browser zoom to 0% the canvas doesn't reset zoom.
Same for the reverse of zooming all the way out and then resetting zoom

It doesn't hurt functionality, just noting it as that was the only thing I found in testing the fix.

Good job on the fix <3
